packages/flutter_test/lib/src/mock_event_channel.dart

```diff
@@ -0,0 +1,59 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/services.dart';
+
+/// A mock stream handler for an [EventChannel] that mimics the native
+/// StreamHandler API.
+abstract class MockStreamHandler {
+  /// Handler for the listen event.
+  void onListen(dynamic arguments, MockStreamHandlerEventSink events);
+
+  /// Handler for the cancel event.
+  void onCancel(dynamic arguments);
+}
+
+/// Convenience class for creating a [MockStreamHandler] inline.
+class InlineMockStreamHandler extends MockStreamHandler {
+  /// Create a new [InlineMockStreamHandler] with the given [onListen] and
+  /// [onCancel] handlers.
+  InlineMockStreamHandler({
+    required void Function(dynamic arguments, MockStreamHandlerEventSink events) onListen,
```

Comment:
fly-by comment: here and elsewhere use typedefs for these function types as per our style guide.